File: angular/src/app/norm/entity/klass.ts
import { Teacher } from "./teacher";

export class Klass {
    id : any;
    name : string;
    teacher : Teacher; 

    constructor( id : any, name : string, teacher : Teacher) {
        this.id = id;
        this.name = name;
        this.teacher = teacher;
    }
}

File: springboot/src/main/java/com/demo/entity/Klass.java
package com.demo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Klass {

    private int id;
    private Teacher teacher;
    private String name;



    @Override
    public String toString() {
        return "Klass{" +
                "id=" + id +
                ", teacher=" + teacher +
                ", name='" + name + '\'' +
                '}';
    }
}
